pytorch_lightning/trainer/distrib_data_parallel.py

```diff
@@ -276,10 +276,10 @@ def ddp_train(self, gpu_idx, model):
         :param cluster_obj:
         :return:
         """
-        # node rank using relative slurm id
-        # otherwise default to node rank 0
+        # node rank using relative slurm id if under slurm management
+        # otherwise use given node rank or default to node rank 0
         try:
-            node_id = os.environ['SLURM_NODEID']
+            node_id = os.environ['SLURM_NODEID'] if self.is_slurm_managing_tasks else os.environ['RANK']
```

Comment:
to make this generic shouldn't we call this ```NODE_RANK```?

Reply:
@williamFalcon RANK is a variable name required by pytorch distributed `env://` initialization method (see [pytorch doc](https://pytorch.org/tutorials/intermediate/dist_tuto.html#initialization-methods))

Reply:
@williamFalcon changed to `NODE_RANK`, since rank is computed inside Pytorch-Lightning based on node rank and gpu count and `RANK` name is misleading.

A section in documentation should also probably be added alongside these changes.